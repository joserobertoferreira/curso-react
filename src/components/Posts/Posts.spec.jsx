/* eslint-disable testing-library/no-node-access */
import { Posts } from '.'

const { render, screen } = require('@testing-library/react')

const props = {
  posts: [
    { id: 1, title: 'Test 1', body: 'Post 1', cover: 'img/cover1.jpg' },
    { id: 2, title: 'Test 2', body: 'Post 2', cover: 'img/cover2.jpg' },
    { id: 3, title: 'Test 3', body: 'Post 3', cover: 'img/cover3.jpg' },
  ],
}

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Posts {...props} />)

    expect(screen.getAllByRole('heading', { name: /Test/i })).toHaveLength(3)
    expect(screen.getAllByRole('heading', { img: /cover/i })).toHaveLength(3)
    expect(screen.getAllByRole(/Post/i)).toHaveLength(3)
    expect(screen.getAllByRole('img', { name: /Test 2/i })).toHaveAttribute('src', 'img/cover2.jpg')
  })

  it('should render posts without posts', () => {
    render(<Posts />)

    expect(screen.queryByRole('heading', { name: /Test/i })).not.toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<Posts {...props} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
