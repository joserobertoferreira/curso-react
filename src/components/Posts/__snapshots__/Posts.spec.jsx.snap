// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Test 1"
      src="img/cover1.jpg"
    />
    <div
      class="post-content"
    >
      <h2>
        Test 1
         
        1
      </h2>
      <p>
        Post 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Test 2"
      src="img/cover2.jpg"
    />
    <div
      class="post-content"
    >
      <h2>
        Test 2
         
        2
      </h2>
      <p>
        Post 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Test 3"
      src="img/cover3.jpg"
    />
    <div
      class="post-content"
    >
      <h2>
        Test 3
         
        3
      </h2>
      <p>
        Post 3
      </p>
    </div>
  </div>
</div>
`;
